Use GNU libiconv's locale_charset() if available because the
native nl_langinfo(CODESET) might not be a canonical name, e.g.,
`646' instead of `ASCII'.

--- lib/toutf8.c.orig	2012-04-27 06:48:08.000000000 -0400
+++ lib/toutf8.c	2012-11-27 15:29:17.000000000 -0500
@@ -58,6 +58,10 @@
 # include <langinfo.h>
 #endif
 
+#if HAVE_ICONV
+extern const char * locale_charset (void);
+#endif
+
 #ifdef _LIBC
 # define stringprep_locale_charset() nl_langinfo (CODESET)
 #else
@@ -89,12 +93,16 @@
   if (charset && *charset)
     return charset;
 
+# if HAVE_ICONV
+  charset = locale_charset();
+# else
 # ifdef HAVE_LANGINFO_CODESET
   charset = nl_langinfo (CODESET);
 
   if (charset && *charset)
     return charset;
 # endif
+# endif
 
   return "ASCII";
 }
